core/src/main/java/org/bptlab/cepta/operators/WeatherLocationCorrelationFunction.java

```diff
@@ -0,0 +1,89 @@
+package org.bptlab.cepta.operators;
+
+import com.github.jasync.sql.db.ConnectionPoolConfigurationBuilder;
+import com.github.jasync.sql.db.QueryResult;
+import com.github.jasync.sql.db.RowData;
+import com.github.jasync.sql.db.pool.ConnectionPool;
+import com.github.jasync.sql.db.postgresql.PostgreSQLConnection;
+import com.github.jasync.sql.db.postgresql.PostgreSQLConnectionBuilder;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import java.util.TreeSet;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.ExecutionException;
+import java.util.function.Supplier;
+import org.apache.flink.api.java.tuple.Tuple2;
+import org.apache.flink.streaming.api.functions.async.ResultFuture;
+import org.apache.flink.streaming.api.functions.async.RichAsyncFunction;
+import org.bptlab.cepta.WeatherData;
+import org.bptlab.cepta.config.PostgresConfig;
+
+public class WeatherLocationCorrelationFunction extends
+    RichAsyncFunction<WeatherData, Tuple2<WeatherData, Integer> > {
+
+  private PostgresConfig postgresConfig = new PostgresConfig();
+
+
+  private transient ConnectionPool<PostgreSQLConnection> connection;
+
+  public WeatherLocationCorrelationFunction(PostgresConfig postgresConfig) {
+    this.postgresConfig = postgresConfig;
+  }
+
+  public void open(org.apache.flink.configuration.Configuration parameters) throws Exception {
```

Comment:
Imo this is too much code sharing. If we keep using the SQL approach we might have to consider extracting into a common base class..